### PR TITLE
SQL: add trace logging for search responses coming from server

### PR DESCRIPTION
**Pull request by astefan**
This PR improves the logging of ES search responses by outputting several meta details, excluding the documents themselves.

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java [hunk 1]
```diff
@@ -150,6 +151,30 @@ public static SearchRequest prepareRequest(Client client, SearchSourceBuilder so
                         includeFrozen ? IndexResolver.FIELD_CAPS_FROZEN_INDICES_OPTIONS : IndexResolver.FIELD_CAPS_INDICES_OPTIONS)
                 .request();
     }
+    
+    protected static void logSearchResponse(SearchResponse response, Logger logger) {
+        List<Aggregation> aggs = Collections.emptyList();
+        if (response.getAggregations() != null) {
```
Comment (bpintea) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java]: Nit: wondering if expanding this branch over the `for` loop wouldn't also simplify (i.e. remove) the in-loop conditional, by allowing to always append `", "` and then strip last two chars outside the loop.

**Review by matriv — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java]: Thanks. I do have a slight preference towards the current approach. I'll merge it as is.

**Pull request merged**